tests/models/data/ddp/train_test_variations.py

```diff
@@ -0,0 +1,54 @@
+"""
+Runs several combinations of `.fit()` and `.test()` on a single node across multiple gpus.
+"""
+from argparse import ArgumentParser
+
+from pytorch_lightning import Trainer, seed_everything
+from tests.base import EvalModelTemplate
+
+
+def variation_fit_test(trainer, model):
+    trainer.fit(model)
+    trainer.test(model)
+
+
+def variation_test_fit(trainer, model):
+    trainer.test(model)
+    trainer.fit(model)
+
+
+def variation_test_test(trainer, model):
+    trainer.test(model)
+    trainer.test(model)
+
+
+def variation_test_fit_test(trainer, model):
+    trainer.test(model)
+    trainer.fit(model)
+    trainer.test(model)
+
+
+def get_variations():
+    variations = [v for v in globals() if v.startswith("variation")]
```

Comment:
This seems kind of hacky, can you just list out the variations please?

Reply:
yes ok, will do

Reply:
if the list becomes long we could also make these functions into a class of methods and collect them there automatically.